cmd/lib.go

```diff
@@ -36,48 +30,58 @@ var defaultStoreHostTargetPath = "/var/lib/resonance"
 
 func wrapHost(ctx context.Context, hst host.Host) (host.Host, error) {
 	var err error
-	if sudo {
+	switch options {
+	case "sudo":
 		hst, err = ihost.NewSudo(ctx, hst)
 		if err != nil {
 			return nil, err
 		}
-	}
-
-	if !disableAgent && ssh != "" {
-		var err error
-		hst, err = ihost.NewAgent(ctx, hst)
-		if err != nil {
-			return nil, err
+	case "sudo,disable-agent", "disable-agent,sudo":
+		if ssh != "" {
+			var err error
+			hst, err = ihost.NewAgent(ctx, hst)
+			if err != nil {
+				return nil, err
+			}
+			hst, err = ihost.NewSudo(ctx, hst)
+			if err != nil {
+				return nil, err
+			}
+		}
+	case "disable-agent":
+		if ssh != "" {
+			var err error
+			hst, err = ihost.NewAgent(ctx, hst)
+			if err != nil {
+				return nil, err
+			}
 		}
+	default:
+		return hst, nil
 	}
 
 	return hst, nil
 }
 
 func addHostFlagsCommon(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(
-		&ssh, "ssh", "", defaultSsh,
+		&ssh, "target-ssh", "s", defaultSsh,
 		"Applies configuration to given hostname using SSH in the format: [<user>[;fingerprint=<host-key fingerprint>]@]<host>[:<port>]",
 	)
 
 	cmd.Flags().StringVarP(
-		&dockerContainer, "docker-container", "", defaultDockerContainer,
-		"Applies configuration to given Docker container name",
+		&docker, "target-docker", "d", defaultDocker,
+		"Applies configuration to given Docker container name \n"+
+			"Use given format 'USER@CONTAINER_ID'",
 	)
-
 	cmd.Flags().StringVarP(
-		&dockerUser, "docker-user", "", defaultDockerUser,
-		"Use given user/group in the format '<name|uid>[:<group|gid>]'",
-	)
-
-	cmd.Flags().BoolVarP(
-		&sudo, "sudo", "", defaultSudo,
-		"Use sudo when interacting with host",
-	)
-
-	cmd.Flags().BoolVarP(
-		&disableAgent, "disable-agent", "", defaultDisableAgent,
-		"Disables copying temporary a small agent to remote hosts. This can make things very slow, as without the agent, iteraction require running multiple commands. The only (unusual) use case for this is when the host architecture is not supported by the agent.",
+		&options, "target-options", "o", defaultDocker,
+		"Usage: --target-options=sudo,disable-agent \n"+
```

Comment:
I'd simplify this to something like: _comma separated list of target options: "sudo", to run as root via sudo; "disable-agent", disable ephemeral agent usage (MUCH slower)_